core/model/edition.py

```diff
@@ -724,6 +724,17 @@ def calculate_author(self):
             sort_author = " ; ".join(sorted(sort_names))
         else:
             sort_author = self.UNKNOWN_AUTHOR
+
+        def truncate_string(mystr: str):
+            if len(mystr) > 1024:
+                return mystr[: (1024 - 3)] + "..."
+            else:
+                return mystr
+
+        # Very long author and sort_author strings can cause the limits of one or more of the
+        # underlying postgres indices.  Therefore,  we limit their size here.
+        author = truncate_string(author)
```

Comment:
Do we need to truncate the `author` value here? Or just the `sort_author`?

Reply:
both author and sort_author are indexed so we'll run into the same problem.

Reply:
I wonder why we index author. It would be nice if we didn't. That way we'd need only truncate the field that's built specifically for sorting.